### PR TITLE
Fix llama-galaxy logit processing when returning logits in generator

### PR DESCRIPTION
**Pull request by tchedaTT**
### Problem description
When llama-galaxy's generator is set to return logits:
- it overrides read_from_device to false
- it doesn't unpad the logits from padded_vocab_size to vocab_size in prefill

Both behaviour were previously unnoticed:
- vLLM v0 always set read_from_device=false
- The padding was ignored in vLLM's sampling pipeline

### What's changed
Removed the override and added missing unpdding

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/docs/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml) CI passes (if applicable)
- [ ] [Galaxy demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by tchedaTT** [comment]
This codepath is not covered by any of the existing galaxy tests, so running galaxy tests seems wasteful. The changes here are picked from tcheda/v1_pipelines_structured_war, where a modified pipeline covering this path ran and passed in https://github.com/tenstorrent/tt-metal/actions/runs/18793135634

**Review by sraizada-tt — APPROVED** [review]

**Review by yalrawwashTT — APPROVED** [review]
